### DISCUSSION TITLE
CTk Theme Builder 2.1

### DISCUSSION THREAD

**Discussion by avalon60** [TomSchimansky/CustomTkinter#1686] [Showcase]
This is a small update, because I needed to get a bug fix out. It does however feature some improvements and a new theme, Harlequin:

![image](https://github.com/TomSchimansky/CustomTkinter/assets/89534395/1b0ee718-f770-4a3a-9dfa-439bc20d9a78)

Improvements:

- Introduced a File -> Provenance menu option - allows you to see who created a theme, when created, last updated and by who etc.
- Smarter rendering of borders to CTkEntry and CTkButton widgets, within the preview panel.
- Improved handling of preview listener timeout.
- New Harlequin theme.

Provenance display:

![image](https://github.com/TomSchimansky/CustomTkinter/assets/89534395/c4bf31a3-fd02-4ca2-ab17-7f3300231129)

Bug fixes:

- Fix shade adjust colour code corruption, where blue channel was incorrectly adjusted,
-Fix a bug whereby CTkScrollableFrame and CTkTabview widget appearances were not updating in the preview panel, when CTkFrame colour property changes were applied.

Download from here:

https://github.com/avalon60/ctk_theme_builder/releases/tag/v2.1.0

New 2.1 wiki entry here:

https://github.com/avalon60/ctk_theme_builder/wiki/CTk-Theme-Builder-2.1-User-Guide

**Comment by Akascape** [comment]
Hi @avalon60, just wanted to inform you that some json theme keywords are changed in Customtkinter 5.2.0 update. 
Basically only 2 changes I noticed: `CTkRadiobutton-->CTkRadioButton` and `CTkCheckbox-->CTkCheckBox`.

**Reply by avalon60** [reply]
Thanks @Akascape. I think that it was me who raised those as issues. I spotted the inconsistencies whilst working on the app.
When did 5.2 come out?

**Reply by Akascape** [reply]
@avalon60 
>When did 5.2 come out?

19th ig.

**Comment by avalon60** [comment]
Closed
